neo/Consensus/ConsensusService.cs

```diff
@@ -153,12 +161,44 @@ private void Log(string message, LogLevel level = LogLevel.Info)
 
         private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
         {
-            if (context.State.HasFlag(ConsensusState.CommitSent))
+            // Node in commit receiving ChangeView should always send the recovery message, to restore
+            // nodes that may have counted their view number past the view that has commit set.
+            bool shouldSendRecovery = context.State.HasFlag(ConsensusState.CommitSent);
+            if (shouldSendRecovery || message.NewViewNumber < context.ViewNumber)
+            {
+                if (!shouldSendRecovery)
+                {
+                    // Limit recovery to sending from `f` nodes when the request is from a lower view number.
+                    int allowedRecoveryNodeCount = context.F;
+                    for (int i = 1; i <= allowedRecoveryNodeCount; i++)
+                    {
+                        if (payload.ValidatorIndex + i % context.Validators.Length != context.MyIndex) continue;
```

Comment:
I thought of a smarter way to handle this. Going to adjust it so that it also takes into affect what `NewViewNumber` is being requested to determine who will respond.

Reply:
From testing I can see that this doesn't work as well as I had hoped because it causes a node requesting restoration each time it counts up its `NewViewNumber` to still pick the same eligible parties to respond, which is not ideal.

Reply:
I've switched to a solution that will switch who the responding validators will be each time the `NewViewNumber` counts  up. It should work great now.